### ISSUE TITLE
使用model.fit训练与使用低级api训练结果差别很大

### ISSUE THREAD

**Issue by y-yunheng** [PaddlePaddle/Paddle#43671]
### 问题描述 Please describe your issue

我使用model.fit训练与使用低级api训练结果差别很大，因为截图不便 所以只截取了前五个epoch的训练 实际上 到了20个epoch的时候 低级api训练的第一个准确率可以到达0.9而model.fit()训练的仅0.49，
现在我只能展示图片，无法直接公开项目。
链接：https://pan.baidu.com/s/1H2xGmjgL_J9pncWHVmgvkg 
提取码：zpgz 
以上是图片描述，请问开发者 model.fit 是否存在特定的训练策略？

**Comment by Liu-xiandong** [comment]
你好，对比了你截图中的训练结果，前5个epoch的训练确实没有太大的差异。目前model.fit()的训练策略可见：https://github.com/PaddlePaddle/Paddle/blob/release/2.3/python/paddle/hapi/model.py#L1557 。与低级api的训练方式并无本质不同。不过，对比了你给出的代码。两者所使用的batch_size有所不同，建议尽可能地将两者对齐再进行对比。

**Comment by y-yunheng** [comment]
> 你好，对比了你截图中的训练结果，前5个epoch的训练确实没有太大的差异。目前model.fit()的训练策略可见：https://github.com/PaddlePaddle/Paddle/blob/release/2.3/python/paddle/hapi/model.py#L1557 。与低级api的训练方式并无本质不同。不过，对比了你给出的代码。两者所使用的batch_size有所不同，建议尽可能地将两者对齐再进行对比。


你好！我查找了原因，因为Accuracy类是我继承paddle.metric.Metric复写的，我用低级API训练时，当一个Epoch结束后，在下面执行测试函数Gmodel.evaluate(self.test_loader)。**但是Gmodel.evaluate并没有在里面执行 acc.reset()** 即准确度没有重置。
我在Gmodel.evaluate(self.test_loader) 上一行添加了acc.reset() 此时model.fit训练与使用低级api训练结果几乎一样了。

**Comment by Liu-xiandong** [comment]
> > 你好，对比了你截图中的训练结果，前5个epoch的训练确实没有太大的差异。目前model.fit()的训练策略可见：https://github.com/PaddlePaddle/Paddle/blob/release/2.3/python/paddle/hapi/model.py#L1557 。与低级api的训练方式并无本质不同。不过，对比了你给出的代码。两者所使用的batch_size有所不同，建议尽可能地将两者对齐再进行对比。
> 
> 你好！我查找了原因，因为Accuracy类是我继承paddle.metric.Metric复写的，我用低级API训练时，当一个Epoch结束后，在下面执行测试函数Gmodel.evaluate(self.test_loader)。**但是Gmodel.evaluate并没有在里面执行 acc.reset()** 即准确度没有重置。 我在Gmodel.evaluate(self.test_loader) 上一行添加了acc.reset() 此时model.fit训练与使用低级api训练结果几乎一样了。

棒棒，找到了原因就好~
